pkg/verification/verify.go

```diff
@@ -285,7 +285,16 @@ func verifyTSRWithChain(ts *timestamp.Timestamp, opts VerifyOpts) error {
 	if len(opts.Roots) == 0 {
 		return fmt.Errorf("no root certificates provided for verifying the certificate chain")
 	}
-
+	rootFound := false
+	for _, cert := range opts.Roots {
+		if cert != nil {
+			rootFound = true
+			break
+		}
+	}
+	if !rootFound {
+		return fmt.Errorf("no valid root certificates provided for verifying the certificate chain - nil opts.Roots")
+	}
 	rootCertPool := x509.NewCertPool()
 	for _, cert := range opts.Roots {
 		rootCertPool.AddCert(cert)
```

Comment:
why not just do a check here to only add the cert to the pool if it is non-nil?

Reply:
but then how we can flag that there are no valid root certificates provided - by comparing `rootCertPool` to an empty pool, something like (using https://pkg.go.dev/crypto/x509#CertPool.Equal)?
```go
if rootCertPool.Equal(x509.NewCertPool()) {
    return fmt.Errorf("no valid root certificates provided for verifying the certificate chain")
}
```

Changed to that in 7ca231502e5bbfe05ea91744e420cefb43886361 - add to the pool only if non-empty and check for `rootCertPool` not Equal to an empty one.